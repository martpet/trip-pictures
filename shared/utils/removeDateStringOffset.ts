export const removeDateStringOffset = (str: string) => {
  // UTC represenation
  if (str.includes('Z')) {
    return str.split('Z')[0];
  }

  // positive offset
  if (str.includes('+')) {
    return str.split('+')[0];
  }

  // negative offset
  if (str.includes('T')) {
    const [time, date] = str.split('T');
    if (date.includes('-')) {
      return `${time}T${date.split('-')[0]}`;
    }
  }

  // no offset or UTC representation
  return str;
};
