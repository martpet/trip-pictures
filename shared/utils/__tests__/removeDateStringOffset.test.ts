import { removeDateStringOffset } from '../removeDateStringOffset';

describe('removeDateStringOffset', () => {
  it('removes UTC representation', () => {
    expect(removeDateStringOffset('2022-11-17T10:08:23.917Z')).toMatchSnapshot();
  });

  it('removes positive offset', () => {
    expect(removeDateStringOffset('2022-11-17T10:08:23.917+02:00')).toMatchSnapshot();
  });

  it('removes negative offset', () => {
    expect(removeDateStringOffset('2022-11-17T10:08:23.917-02:00')).toMatchSnapshot();
  });

  it('returns original string if no offset representation', () => {
    expect(removeDateStringOffset('2022-11-17T10:08:23.917')).toMatchSnapshot();
  });
});
