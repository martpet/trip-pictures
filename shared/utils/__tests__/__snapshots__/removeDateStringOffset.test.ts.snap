// Vitest Snapshot v1

exports[`removeDateStringOffset > removes UTC representation 1`] = `"2022-11-17T10:08:23.917"`;

exports[`removeDateStringOffset > removes negative offset 1`] = `"2022-11-17T10:08:23.917"`;

exports[`removeDateStringOffset > removes positive offset 1`] = `"2022-11-17T10:08:23.917"`;

exports[`removeDateStringOffset > returns original string if no offset representation 1`] = `"2022-11-17T10:08:23.917"`;
